queries.sql

```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
 
 SELECT *
 FROM animals
-WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'unspecified';
+
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS `unspecified`;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals;
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals
+WHERE date_of_birth > '2021-01-01';
+
+SAVEPOINT deleted_data_with_date_2021_01_01;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SAVEPOINT deleted_data_with_date_2021_01_01;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
```

Comment:
- [ ] Please remove these lines as per the requirement. There should be only 'COMMIT' after this query.
![image](https://github.com/Peter-Dumbari/SQL_database/assets/91486791/02da5c2c-0ffe-4d5b-ae87-808fbfbf44b8)